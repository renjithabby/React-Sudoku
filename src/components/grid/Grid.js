import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Cell from '../cell/Cell';
import './style.css';


class Grid extends Component {

    handleCellClick = (event)=>{
        this.props.assignValueToSelectedCell(parseInt(event.target.id));
    }

    getGridCells =(gridData)=>{
        let  gridCells =  [];
        gridData.forEach(gridRow => {
            gridCells = [...gridCells, gridRow.map(({id, value,invalid,prefilled}) => <Cell handleCellClick={this.handleCellClick} key={id} id ={id} value={value} invalid ={invalid} prefilled ={prefilled}  />)];
        });
        return gridCells;
    }

    componentDidMount(){
        this.props.generateGridData();
    }

    render(){
        const {gridData} = this.props;
        return (
            <div className="grid-container">
                {
                  [...this.getGridCells(gridData)]
                }
            </div>

        );
    };

}

Grid.propTypes= {
    gridData : PropTypes.array.isRequired,
    generateGridData : PropTypes.func.isRequired,
    assignValueToSelectedCell : PropTypes.func.isRequired,
}

export default Grid;
