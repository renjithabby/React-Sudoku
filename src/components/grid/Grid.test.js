import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Grid from './Grid';

configure({adapter : new Adapter()});

describe.only('<Grid />', () => {
    
    const gridData =[[
        {
            id:0,
            value:1,
            invalid: false,
            prefilled:false
        },
        {
            id:1,
            value:2,
            invalid: true,
            prefilled:true
        }

    ]]

    it('It matches the snapshot with props dataGrid value', () => {
        const tree = shallow(<Grid  gridData={gridData} generateGridData={()=>{}} assignValueToSelectedCell={()=>{}} />);
        expect(toJson(tree)).toMatchSnapshot();
    });

})