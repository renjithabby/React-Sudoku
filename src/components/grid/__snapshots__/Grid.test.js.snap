// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> It matches the snapshot with props dataGrid value 1`] = `
<div
  className="grid-container"
>
  <Cell
    handleCellClick={[Function]}
    id={0}
    invalid={false}
    key="0"
    prefilled={false}
    value={1}
  />
  <Cell
    handleCellClick={[Function]}
    id={1}
    invalid={true}
    key="1"
    prefilled={true}
    value={2}
  />
</div>
`;
